test/validation/dry_validation_test.rb

```diff
@@ -95,7 +97,7 @@ class AlbumForm < Reform::Form
         end
       end
 
-      validation :default do
+      validation :default, with: { title: -> { title }, guitar: Guitar } do
```

Comment:
@apotonick , as @solnic [said](https://github.com/apotonick/reform/pull/347#issuecomment-196900284), injecting form instance _every_ time costs us rebuilding the whole schema. Here's the [implementation](https://github.com/dry-rb/dry-validation/blob/master/lib/dry/validation/schema.rb#L179) . That's why I've prepared a logic that let's one decide, whether an arbitrary dependency is needed.